aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -441,11 +443,77 @@ void uniform_kernel_cuda(TensorIterator& iter, double from_, double to_, Generat
    });
 }
 
+void random_kernel_cuda(TensorIterator& iter, uint64_t range, int64_t base, Generator* gen_) {
+  auto gen = check_generator<CUDAGenerator>(gen_, &globalContext().defaultGenerator(kCUDA));
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, iter.dtype(), "random_cuda", [&] {
+    using accscalar_t = at::acc_type<scalar_t, true>;
+    if (std::is_same<scalar_t, double>::value || std::is_same<scalar_t, int64_t>::value) {
```

Comment:
nit: Consider using template specializations to do this in the future. You're generating dead code when you do it this way; fortunately that code type checks and will get DCE'd later... but best not to generate it at all in the first place.